### ISSUE TITLE
Error telemetry: Max call stack size exceeded.

### ISSUE THREAD

**Issue by judej** [microsoft/pyright#6178]
Seeing validateFunctionArgumentTypes in most traces. Couple of hundred machines a week. Don't have a repro. Seeing it on Pylance 2023.10.20.

`
RangeError: Maximum call stack size exceeded
at evaluator /src/analyzer/constraintSolver.ts:77:4
at evaluatorInterface /src/analyzer/typeEvaluator.ts:22297:28
at assignType /src/analyzer/typeEvaluator.ts:11747:24
at validateArgType /src/analyzer/typeEvaluator.ts:11206:30
at func (<anonymous>)
at forEach /src/analyzer/typeEvaluator.ts:11205:31
at validateFunctionArgumentTypes /src/analyzer/typeEvaluator.ts:11645:23
at errorNode /src/analyzer/typeEvaluator.ts:11262:16
at validateFunctionArgumentTypes /src/analyzer/typeEvaluator.ts:11645:23
at errorNode /src/analyzer/typeEvaluator.ts:11262:16
at validateFunctionArgumentTypes /src/analyzer/typeEvaluator.ts:11645:23
at errorNode /src/analyzer/typeEvaluator.ts:11262:16
at validateFunctionArgumentTypes /src/analyzer/typeEvaluator.ts:11645:23
at errorNode /src/analyzer/typeEvaluator.ts:11262:16
at validateFunctionArgumentTypes /src/analyzer/typeEvaluator.ts:11645:23
at errorNode /src/analyzer/typeEvaluator.ts:11262:16
at validateFunctionArgumentTypes /src/analyzer/typeEvaluator.ts:11645:23
at errorNode /src/analyzer/typeEvaluator.ts:11262:16
at validateFunctionArgumentTypes /src/analyzer/typeEvaluator.ts:11645:23
at errorNode /src/analyzer/typeEvaluator.ts:11262:16
at validateFunctionArgumentTypes /src/analyzer/typeEvaluator.ts:11645:23
at errorNode /src/analyzer/typeEvaluator.ts:11262:16
at validateFunctionArgumentTypes /src/analyzer/typeEvaluator.ts:11645:23
at validateFunctionArgumentsForParamSpecSignature /src/analyzer/typeEvaluator.ts:11564:19
`

**Comment by erictraut** [comment]
Based on the stack trace, my best guess is that this was introduced as part of [this change](https://github.com/microsoft/pyright/commit/b688ebb1ae5f203b458df0a06aba05ed131f775e).

Without a repro, this may be difficult to diagnose. I'll think about possible ways to gather more clues.

**Comment by rslinckx** [comment]
@erictraut  Here is a reproducer with 1.330, 1.331

Working with version 1.329

Do not try to make sense of the code, it has been reduced down from a much larger actual file, but this is the minimum required code to make pyright crash with a maximum recursion

```
from typing import Any, Generic, ParamSpec
P = ParamSpec("P")

class TaskDeclaration(Generic[P]):
    pass

class Dispatcher:
    def dispatch(self, task_declaration: TaskDeclaration[P], /, *args: P.args, **kwargs: P.kwargs) -> None:
        pass

class QueueManager:
    dispatcher: Dispatcher

    def foo(self, stub: TaskDeclaration[P]) -> Any:
        def decorate(func: Any) -> Any:
            def dispatch(*args: P.args, **kwargs: P.kwargs) -> None:
                self.dispatcher.dispatch(stub, *args, **kwargs)

            return stub

        return decorate
```

**Comment by erictraut** [comment]
@rslinckx, thanks for the repro, that was very helpful.

I've put a short-term fix in place that addresses the crash. It's not a complete fix because it allows for false negatives in some cases. I'll leave this issue open until I plug those holes.

**Comment by rslinckx** [comment]
1.332 seems to work for me!

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.334, which I just published. It will also be included in a future release of pylance.
